src/Corax/IndexEntry/IndexEntryWriter.cs

```diff
@@ -105,13 +105,128 @@ public void Write(int field, ReadOnlySpan<byte> value)
             _dataIndex += sizeof(IndexEntryFieldType);
             return;
         }
-        
+
         int length = VariableSizeEncoding.Write(buffer, value.Length, _dataIndex);
         _dataIndex += length;
         value.CopyTo(buffer.Slice(_dataIndex, value.Length));
         _dataIndex += value.Length;
     }
 
+    public void WriteNullDynamic(string name)
+    {
+        using var _ = _context.From(name, ByteStringType.Immutable, out var fieldNameStr);
+        Span<byte> tmpBuffer = stackalloc byte[10];
+        int fieldLenLen = VariableSizeEncoding.Write(tmpBuffer, fieldNameStr.Length);
+
+        var requiredSize = fieldLenLen + fieldNameStr.Length;
+        
+        if (FreeSpace < requiredSize)
+            UnlikelyGrowAuxiliaryBuffer(requiredSize);
+        
+        Unsafe.WriteUnaligned(ref Buffer[_dataIndex], IndexEntryFieldType.Null);
+        _dataIndex += sizeof(IndexEntryFieldType);
+    }
+    
+    public void WriteSpatialDynamic(string name, ReadOnlySpan<CoraxSpatialPointEntry> entries)
+    {
+        if (entries.Length == 0)
+            return;
+
+        ComputeSpaceRequirements(entries, out int requiredSize, out int maxGeohashLength);
+        
+        using var _ = _context.From(name, ByteStringType.Immutable, out var fieldNameStr);
+        Span<byte> tmpBuffer = stackalloc byte[10];
+        int fieldLenLen = VariableSizeEncoding.Write(tmpBuffer, fieldNameStr.Length);
+
+        requiredSize += fieldLenLen + fieldNameStr.Length;
+        
+        if (FreeSpace < requiredSize)
+            UnlikelyGrowAuxiliaryBuffer(requiredSize);
+        WriteDynamicFieldName(fieldNameStr);
```

Comment:
Why are we writing the dynamic field name? Didn't you say it was not going to be stored at the entry level? I though that we would have a tree that would map those to actual `Id`s ... In fact, if we do that, then we can address directly using negative numbers instead of positive to know immediately if we have a dynamic or a static field.

Reply:
That would also save a lot of space, since we can use a `CompactTree` to store the assigned `FieldId`.

Reply:
That isn't a good idea, because it's *really* common to have totally unique field names.
Imagine things like: `Guid`, `DateTime`, etc as field names.

A `CompactTree` for this would require that we'll need to track usage count for dynamic fields (to allow removing it) and as well as the _inverse_ mapping, from dynamic id to the field name.